File: 2350-shortest-impossible-sequence-of-rolls/2350-shortest-impossible-sequence-of-rolls.java
class Solution {
    public int shortestSequence(int[] rolls, int k) {
        Set<Integer> seen = new HashSet<>();
        int ans = 1;
        for(int roll : rolls){
            seen.add(roll);
            
            if(seen.size() == k){
                seen.clear();
                ans += 1;
            }
        }
        return ans;
    }
}

File: 2350-shortest-impossible-sequence-of-rolls/2350-shortest-impossible-sequence-of-rolls.py
class Solution:
    def shortestSequence(self, rolls: List[int], k: int) -> int:
        seen = set()
        ans = 1
        for roll in rolls:
            seen.add(roll)
            
            if len(seen) == k:
                seen = set()
                ans += 1
        return ans